tests/ui/lint/unused_parens_cast.rs

```diff
@@ -0,0 +1,9 @@
+// check-pass
+
+#![warn(unused_parens)]
+
+fn hello(f: f64) {}
+
+fn main() {
+    hello((1.0f32) as f64) // ~WARN unnecessary parentheses
```

Comment:
Can you maybe add more test cases here that test [this](https://github.com/rust-lang/rust/blob/9541afaa167846ec7bee04a9851346d964fc60c1/compiler/rustc_lint/src/unused.rs#L574-L584) more elabortately

Reply:
added more testcases :)